modules/apps/object/object-api/src/main/java/com/liferay/object/field/util/ObjectFieldUtil.java

```diff
@@ -282,17 +314,40 @@ else if (Objects.equals(
 	}
 
 	private static void _validateNewValue(
-			String dbType, Object existingValue, String objectFieldName,
-			Object value)
+			Object existingValue, ObjectField objectField, Object value)
 		throws PortalException {
 
 		if (Validator.isNull(existingValue) && Validator.isNull(value)) {
 			return;
 		}
 
-		if (Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_BLOB) ||
-			Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_CLOB) ||
-			Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_STRING)) {
+		if (Objects.equals(
```

Comment:
Just to let you know that you can also use `ObjectFieldImpl#compareBusinessType` for this.

This change is not mandatory, but if you want, you can change it here or send a follow-up PR.